src/ol_infrastructure/applications/forum/__main__.py

```diff
@@ -0,0 +1,235 @@
+"""Create the resources needed to run a forum server.
+"""
+import base64
+import json
+import textwrap
+from pathlib import Path
+
+import pulumi_vault as vault
+import yaml
+from pulumi import Config, StackReference
+from pulumi_aws import ec2, get_caller_identity, iam
+
+from bridge.lib.magic_numbers import FORUM_SERVICE_PORT
+from bridge.secrets.sops import read_yaml_secrets
+from ol_infrastructure.components.aws.auto_scale_group import (
+    BlockDeviceMapping,
+    OLAutoScaleGroupConfig,
+    OLAutoScaling,
+    OLLaunchTemplateConfig,
+    TagSpecification,
+)
+from ol_infrastructure.lib.aws.ec2_helper import InstanceTypes, default_egress_args
+from ol_infrastructure.lib.consul import get_consul_provider
+from ol_infrastructure.lib.ol_types import AWSBase
+from ol_infrastructure.lib.pulumi_helper import parse_stack
+from ol_infrastructure.lib.vault import setup_vault_provider
+
+stack_info = parse_stack()
+forum_config = Config("forum")
+if Config("vault").get("address"):
+    setup_vault_provider()
+network_stack = StackReference(f"infrastructure.aws.network.{stack_info.name}")
+policy_stack = StackReference("infrastructure.aws.policies")
+dns_stack = StackReference("infrastructure.aws.dns")
+consul_stack = StackReference(
+    f"infrastructure.consul.{stack_info.env_prefix}.{stack_info.name}"
+)
+kms_stack = StackReference(f"infrastructure.aws.kms.{stack_info.name}")
+vault_stack = StackReference(f"infrastructure.vault.operations.{stack_info.name}")
+print(f"applications.edxapp.{stack_info.env_prefix}.{stack_info.name}")
+edxapp = StackReference(
+    f"applications.edxapp.{stack_info.env_prefix}.{stack_info.name}"
+)
+
+mitodl_zone_id = dns_stack.require_output("odl_zone_id")
+env_name = f"{stack_info.env_prefix}-{stack_info.env_suffix}"
+target_vpc_name = forum_config.get("target_vpc")
+target_vpc = network_stack.require_output(target_vpc_name)
+
+aws_account = get_caller_identity()
+vpc_id = target_vpc["id"]
+forum_server_ami = ec2.get_ami(
+    filters=[
+        ec2.GetAmiFilterArgs(name="name", values=["forum-server-*"]),
+        ec2.GetAmiFilterArgs(name="virtualization-type", values=["hvm"]),
+        ec2.GetAmiFilterArgs(name="root-device-type", values=["ebs"]),
+    ],
+    most_recent=True,
+    owners=[aws_account.account_id],
+)
+aws_config = AWSBase(
+    tags={
+        "OU": forum_config.require("business_unit"),
+        "Environment": env_name,
+        "Application": "forum",
```

Comment:
To make things more explicit we should probably use `open-edx-forum` everywhere so that there's no confusion if we have an app in the future with some level of forum functionality.